src/io.rs

```diff
@@ -469,17 +470,15 @@ fn begin_filling(iobuffer: IoBuffer, fd: OwnedFd) -> FdMonitorItemId {
     // In this case, when complete_background_fillthread() is called, we grab the file descriptor
     // and read until we get EAGAIN and then give up.
     // Run our function to read until the receiver is closed.
-    let item_callback: Callback = Box::new(move |fd: &mut AutoCloseFd| {
-        assert!(fd.as_raw_fd() >= 0, "Invalid fd");
+    let item_callback: Callback = Box::new(move |fd: &mut Option<OwnedFd>| {
         let mut buf = iobuffer.0.lock().unwrap();
-        let ret = IoBuffer::read_once(fd.as_raw_fd(), &mut buf);
+        let ret = IoBuffer::read_once(fd.as_ref().unwrap().as_raw_fd(), &mut buf);
         if ret == 0 || (ret < 0 && ![EAGAIN, EWOULDBLOCK].contains(&errno::errno().0)) {
             // Either it's finished or some other error - we're done.
-            fd.close();
+            drop(fd.take());
```

Comment:
Hmm maybe `fd.take()` is better -- the `drop(..)` doesn't change behavior and I'm not sure if the extra explicitness is needed.

Reply:
For clarity, I prefer using an explicit `drop` rather than ignoring the return value, which would otherwise trigger the `unused_results` lint.